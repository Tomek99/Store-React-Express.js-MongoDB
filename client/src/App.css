@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;500;700&display=swap");

* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  outline: none;
  border: none;
  text-decoration: none;
  font-family: "Roboto", sans-serif;
  /* transition: 0.2s linear; */
}

:root {
  --main-color: #d3ad7f;
  --main-color-darken: #d3a369;
  --black: #0e0e13;
  --light-black: #1a1a23;
  --bg: #04040b;
  --blue: #151522;
  --grey: #323232;
  --grey-darken: #202029;
  --green: rgb(31, 129, 20);
  --green-darken: rgb(27, 113, 17);
  --border: 0.1rem solid rgba(255, 255, 255, 0.3);
  --red: rgb(180, 3, 28);
  --red-darken: rgb(156, 4, 24);
  --shadow: rgba(255, 255, 255, 0.491) 0px 2px 4px 0px,
    rgba(255, 255, 255, 0.496) 0px 0px 2px 1px;
  --shadow-thin: rgba(255, 255, 255, 0.25) 0px 0px 1px 1px,
    rgba(255, 255, 255, 0.13) 0px 0px 1px 1px;
  --blur-screen: rgb(60, 60, 60);
}

html {
  font-size: 62.5%;
  overflow-x: hidden;
  scroll-padding-top: 9rem;
  scroll-behavior: smooth;
}

body {
  background: var(--bg);
}

#root {
  display: grid;
  min-height: 100vh;
  min-height: 100dvh;
  grid-template-rows: auto 1fr auto;
}
section.columnWeb {
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 30px;
}

section.adminWeb {
  display: flex;
  justify-content: stretch;
}

section.columnAdmin {
}

/* width */
::-webkit-scrollbar {
  width: 5px;
}

/* Track */
::-webkit-scrollbar-track {
  background: var(--bg);
}

/* Handle */
::-webkit-scrollbar-thumb {
  background: var(--main-color);
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: #555;
}

.Toastify__toast-body {
  font-size: 1.4rem;
}

.absoluteDivApp {
  position: absolute;
}
