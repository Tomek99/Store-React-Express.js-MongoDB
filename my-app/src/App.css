@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;500;700&display=swap');

* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  outline: none;
  border: none;
  text-decoration: none;
  text-transform: capitalize;
  transition: .2s linear;
  font-family: 'Roboto', sans-serif;
}

:root {
  --main-color: #d3ad7f;
  --black: #13131a;
  --bg: #010103;
  --border: .1rem solid rgba(255, 255, 255, .3);
}

html {
  font-size: 62.5%;
  overflow-x: hidden;
  scroll-padding-top: 9rem;
  scroll-behavior: smooth;
}

body {
  background: var(--bg);
}

section.columnWeb {
  display: flex;
  flex-direction: column;
  gap: 20px 0px;
}