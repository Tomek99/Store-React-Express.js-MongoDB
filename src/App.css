@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;500;700&display=swap");

* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  outline: none;
  border: none;
  text-decoration: none;
  transition: 0.3s linear; /* DO usuniÄ™cia */
  text-transform: capitalize;
  font-family: "Roboto", sans-serif;
}

:root {
  --main-color: #d3ad7f;
  --black: #13131a;
  --bg: #010103;
  --border: 0.1rem solid rgba(255, 255, 255, 0.3);
}

html {
  font-size: 62.5%;
  overflow-x: hidden;
  scroll-padding-top: 9rem;
  scroll-behavior: smooth;
}

body {
  background: var(--bg);
  overflow: hidden;
}

section.columnWeb {
  display: flex;
  flex-direction: column;
  gap: 20px 0px;
}

/* width */
::-webkit-scrollbar {
  width: 10px;
}

/* Track */
::-webkit-scrollbar-track {
  background: var(--bg);
}

/* Handle */
::-webkit-scrollbar-thumb {
  background: var(--main-color);
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: #555;
}
