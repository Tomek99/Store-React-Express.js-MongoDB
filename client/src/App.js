import "./App.css";
import React, { useState, useEffect } from "react";
import MessengerCustomerChat from "react-messenger-customer-chat";
import axios from "axios";
import { Context } from "./Contexts/Context";
import { Route, Routes, Navigate, useLocation } from "react-router-dom";
import {
  Home,
  Articles,
  Products,
  Reviews,
  AboutUs,
  Account,
  Wish,
} from "./pages";
import {
  NavigationBar,
  NavigationBarOrder,
  ContactSection,
  Footer,
  FooterOrder,
  ProductDetails,
  Orders,
  Order,
  UserReviews,
  Address,
  Settings,
  ReturnComplaint,
  LogIn,
  SignUp,
  RecoverPassword,
  Protected,
  ViewCart,
  AccountContent,
  OrderDetails,
  Summary,
  PaymentSuccess,
  PaymentCanceled,
  ProtectedOrder,
} from "./components";

function App() {
  /*----------- products ----------- */
  const [products, setProducts] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    setLoading(true);
    const fetchData = async () => {
      const products = await axios.get(
        `${process.env.REACT_APP_API_URI}/api/products`
      );
      setLoading(false);
      setProducts(products.data);
    };
    fetchData();
  }, []);

  /*----------- login ----------- */
  const [isLogIn, setIsLogIn] = useState(() => {
    if (localStorage.getItem("is-logged") !== null) {
      const storedValue = localStorage.getItem("is-logged");
      return storedValue === "true" ? true : false;
    } else return false;
  });

  const [user, setUser] = useState(() => {
    const storedValue = localStorage.getItem("user-data");
    if (storedValue !== null && isLogIn) return JSON.parse(storedValue);
    else return {};
  });

  function logIn(data) {
    setIsLogIn(true);
    setUser(data);
    localStorage.clear(); // TEST
    localStorage.setItem("user-data", JSON.stringify(data));
    localStorage.setItem("is-logged", true);
  }

  function logOut() {
    setIsLogIn(false);
    setUser({});
    localStorage.clear();
  }

  /*----------- cart ----------- */
  const [cartItems, setCartItems] = useState(() => {
    const storedValue = localStorage.getItem("cart");
    if (storedValue !== null) return JSON.parse(storedValue);
    else return [];
  });
  const [cartValue, setCartValue] = useState(() => {
    const storedValue = localStorage.getItem("cart-value");
    if (storedValue !== null) return JSON.parse(storedValue);
    else return 0;
  });
  const [cartSave, setCartSave] = useState(() => {
    const storedValue = localStorage.getItem("cart-save");
    if (storedValue !== null) return JSON.parse(storedValue);
    else return 0;
  });

  const [cartQuantity, setCartQuantity] = useState(() => {
    const storedValue = localStorage.getItem("cart-quantity");
    if (storedValue !== null) return JSON.parse(storedValue);
    else return 0;
  });

  function addItem(item) {
    const newItemIndex = cartItems.findIndex(
      (element) => element._id === item._id
    );

    if (newItemIndex === -1) {
      setCartItems([...cartItems, { ...item }]);
    } else {
      let newArr = [...cartItems];

      newArr[newItemIndex].quantity =
        newArr[newItemIndex].quantity + item.quantity;

      setCartItems(newArr);
    }

    setCartQuantity(cartQuantity + item.quantity);
    appendPrice(item.price, item.oldPrice, item.quantity);
  }

  function deleteItem(id, newPrice, oldPrice) {
    const cartList = cartItems.filter((item) => item._id !== id);
    const findItem = cartItems.filter((item) => item._id === id);

    setCartItems(cartList);
    setCartQuantity(cartQuantity - findItem[0].quantity);
    subtractPrice(newPrice, oldPrice, findItem[0].quantity);
  }

  function appendPrice(newPrice, oldPrice, quantity) {
    setCartValue(
      Math.round((cartValue + parseFloat(newPrice) * quantity) * 100) / 100
    );

    if (oldPrice !== 0) {
      setCartSave(
        Math.round(
          (cartSave +
            (parseFloat(oldPrice) - parseFloat(newPrice)) * quantity) *
            100
        ) / 100
      );
    }
  }

  function subtractPrice(newPrice, oldPrice, quantity) {
    setCartValue(
      Math.round((cartValue - parseFloat(newPrice) * quantity) * 100) / 100
    );

    if (oldPrice !== 0) {
      setCartSave(
        Math.round(
          (cartSave -
            (parseFloat(oldPrice) - parseFloat(newPrice)) * quantity) *
            100
        ) / 100
      );
    }
  }

  function clearTheCart() {
    setCartItems([]);
    setCartQuantity(0);
    setCartValue(0);
    setCartSave(0);

    localStorage.setItem("cart", JSON.stringify([]));
    localStorage.setItem("cart-quantity", 0);
    localStorage.setItem("cart-value", 0);
    localStorage.setItem("cart-save", 0);
  }
  useEffect(() => {
    localStorage.setItem("cart-save", cartSave);
    localStorage.setItem("cart-value", cartValue);
    localStorage.setItem("cart-quantity", cartQuantity);
    localStorage.setItem("cart", JSON.stringify(cartItems));
  }, [cartValue, cartQuantity, cartItems, cartSave]);

  /*----------- wishList ----------- */
  const [wishList, setWishList] = useState([]);

  function addItemWishList(item) {
    setWishList([...wishList, { ...item }]);
  }

  /*----------- order ----------- */
  const [order, setOrder] = useState(() => {
    const storedValue = localStorage.getItem("order");
    if (storedValue !== null) return JSON.parse(storedValue);
    else return {};
  });

  function addOrder(order) {
    const address = {
      name: order.name,
      street: order.street,
      house: order.house,
      ZIP_code: order.ZIP_code,
      city: order.city,
      number: order.number,
      email: order.email,
    };

    const company = {
      companyNip: order.companyNip,
      companyName: order.companyName,
      companyStreet: order.companyStreet,
      companyZIPcode: order.companyZIPcode,
      companyCity: order.companyCity,
    };

    let invoice = {
      name: order.i_name,
      street: order.i_street,
      ZIP_code: order.i_ZIP_code,
      city: order.i_city,
    };
    if (order.i_name) {
      invoice = {
        name: order.i_name,
        street: order.i_street,
        ZIP_code: order.i_ZIP_code,
        city: order.i_city,
      };
    } else {
      invoice = {
        name: order.name,
        street: order.street,
        ZIP_code: order.ZIP_code,
        city: order.city,
      };
    }
    const orderUpdate = {
      // userOrdersId: user.orders,
      userOrdersId: user.orders,
      save: cartSave,
      cartValue: cartValue,
      totalCost:
        cartValue +
        JSON.parse(order.deliveryFee) +
        JSON.parse(order.paymentFee),
      supplyPrice: 15,
      products: cartItems,
      address: address,
      company: company,
      invoice: invoice,
      payment: order.payment,
      paymentFee: order.paymentFee,
      delivery: order.delivery,
      deliveryFee: order.deliveryFee,
      shopper: order.shopper,
      comment: order.comment,
      activeAddress: order.activeAddress,
      activeCompany: order.activeCompany,
      activeInvoice: order.activeInvoice,
      activeComment: order.activeComment,
    };

    setOrder(orderUpdate);
    localStorage.setItem("order", JSON.stringify(orderUpdate));
  }

  /*----------- location ----------- */
  const location = useLocation();

  /*----------- params ----------- */

  /*----------- navigate on Summary ----------- */
  const [isUserNavigateToSummary, setIsUserNavigateToSummary] = useState(false);

  function handleUserNavigateToSummary() {
    setIsUserNavigateToSummary(true);
  }

  return (
    <Context.Provider
      value={{
        addItem,
        deleteItem,
        addItemWishList,
        logIn,
        logOut,
        clearTheCart,
        addOrder,
        isLogIn,
        cartItems,
        cartValue,
        cartSave,
        cartQuantity,
        products,
        user,
        loading,
        order,
      }}
    >
      {(() => {
        switch (location.pathname) {
          case "/order":
          case "/order/summary":
          case "/order/success":
          case "/order/canceled":
            return <NavigationBarOrder />;

          default:
            return <NavigationBar />;
        }
      })()}
      <section className="columnWeb">
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="about-us" element={<AboutUs />} />
          <Route path="products" element={<Products />} />
          <Route path="products/:id" element={<ProductDetails />} />
          <Route path="reviews" element={<Reviews />} />
          <Route path="contact" element={<ContactSection />} />
          <Route path="articles" element={<Articles />} />
          <Route path="blog" element={<Articles />} />

          <Route
            path="log-in"
            element={
              <Protected isLogIn={!isLogIn} navigate="/">
                <LogIn />
              </Protected>
            }
          />
          <Route
            path="sign-up"
            element={
              <Protected isLogIn={!isLogIn} navigate="/">
                <SignUp />
              </Protected>
            }
          />
          <Route
            path="recover-password"
            element={
              <Protected isLogIn={!isLogIn} navigate="/">
                <RecoverPassword />
              </Protected>
            }
          />

          <Route
            path="/"
            element={
              <Protected isLogIn={isLogIn} navigate="/log-in">
                <Account />
              </Protected>
            }
          >
            <Route path="account" element={<AccountContent />} />
            <Route path="orders" element={<Orders />} />
            {/* <Route path="orders/:id" element={<OrderDetails />} /> */}
            <Route path="returns" element={<ReturnComplaint />} />
            <Route path="user-reviews" element={<UserReviews />} />
            <Route path="address" element={<Address />} />
            <Route path="settings" element={<Settings />} />
          </Route>

          <Route
            path="order"
            element={
              <ProtectedOrder
                isLogIn={isLogIn}
                cartItemsLen={cartItems.length}
                navigate="/log-in"
              >
                <Order
                  handleUserNavigateToSummary={handleUserNavigateToSummary}
                />
              </ProtectedOrder>
            }
          />

          <Route
            path="order/summary"
            element={
              <Protected isLogIn={isUserNavigateToSummary} navigate="/order">
                <Summary />
              </Protected>
            }
          />

          <Route
            path="order/success"
            element={<PaymentSuccess clearTheCart={clearTheCart} />}
          />
          <Route path="order/canceled" element={<PaymentCanceled />} />

          <Route path="wish-list" element={<Wish />} />
          <Route path="cart" element={<ViewCart />} />
          <Route path="*" element={<Navigate to="/" replace />} />
        </Routes>
        {/* <MessengerCustomerChat pageId="" appId="" htmlRef="" /> */}
      </section>
      {(() => {
        switch (location.pathname) {
          case "/order":
          case "/order/summary":
          case "/cart":
          case "/order/success":
          case "/order/canceled":
            return <FooterOrder />;

          default:
            return <Footer />;
        }
      })()}
    </Context.Provider>
  );
}

export default App;
