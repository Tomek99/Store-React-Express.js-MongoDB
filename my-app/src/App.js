import './App.css';
import HeaderSection from './components/HeaderSection/HeaderSection';
import HomeSection from './components/HomeSection/HomeSection';
import AboutSection from './components/AboutSection/AboutSection';
import MenuSection from './components/MenuSection/MenuSection';
import ProductsSection from './components/ProductsSection/ProductsSection';
import ReviewSection from './components/ReviewSection/ReviewSection';
import ContactSection from './components/ContactSection/ContactSection';
import BlogsSection from './components/BlogsSection/BlogsSection';
import FooterSection from './components/FooterSection/FooterSection';

function App() {
    return (
        <section className='columnWeb'>
            <HeaderSection />
            <HomeSection />
            <AboutSection />
            <MenuSection />
            <ProductsSection />
            <ReviewSection />
            <ContactSection />
            <BlogsSection />
            <FooterSection />
        </section>
    );
}

export default App;
