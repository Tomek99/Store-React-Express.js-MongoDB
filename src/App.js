import "./App.css";
import React, { useEffect, useState } from "react";
import { Route, Routes } from "react-router-dom";
import {
  Home,
  Articles,
  Products,
  Menu,
  Reviews,
  AboutUs,
} from "./pages/index";

import { NavigationBar, ContactSection, Footer } from "./components/index";
function App() {
  const [basketItems, setBasketItem] = useState([]);
  const [basketPrice, setBasketPrice] = useState({ currentPrice: 0, save: 0 });

  const addItem = (item) => {
    if (isRepeat(item.idProduct)) setBasketItem([...basketItems, item]);

    setBasketPrice((prevPrice) => ({
      ...prevPrice,
      currentPrice:
        Math.round((prevPrice.currentPrice + item.newPrice) * 100) / 100,
      save:
        Math.round((prevPrice.save + (item.oldPrice - item.newPrice)) * 100) /
        100,
    }));
  };

  const deleteItem = (id, newPrice, oldPrice) => {
    const basketList = basketItems.filter((item) => item.idProduct !== id);
    const findItem = basketItems.filter((item) => item.idProduct === id);

    setBasketItem(basketList);

    setBasketPrice((prevPrice) => ({
      ...prevPrice,
      currentPrice:
        Math.round(
          (prevPrice.currentPrice - newPrice * findItem[0].quantity) * 100
        ) / 100,
      save:
        Math.round(
          (prevPrice.save - (oldPrice - newPrice) * findItem[0].quantity) * 100
        ) / 100,
    }));
  };

  const isRepeat = (idProduct) => {
    for (let i = 0; i < basketItems.length; i++) {
      if (basketItems[i].idProduct === idProduct) {
        basketItems[i].quantity += 1;
        return false;
      }
    }
    return true;
  };

  return (
    <section className="columnWeb">
      <NavigationBar
        basketItems={basketItems}
        deleteItem={deleteItem}
        basketPrice={basketPrice}
      />
      <Routes>
        <Route
          path="/"
          element={
            <Home
              addItem={addItem}
              basketItems={basketItems}
              basketPrice={basketPrice}
            />
          }
        />
        {<Route path="/about-us/*" element={<AboutUs />} />}
        <Route path="/menu" element={<Menu addItem={addItem} />} />
        <Route path="/products" element={<Products addItem={addItem} />} />
        <Route path="/reviews" element={<Reviews />} />
        <Route path="/contact" element={<ContactSection />} />
        <Route path="/articles" element={<Articles />} />
        <Route path="/blog" element={<Articles />} />
      </Routes>

      <Footer />
    </section>
  );
}

export default App;
