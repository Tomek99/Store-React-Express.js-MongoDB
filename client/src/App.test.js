import { render, screen } from "@testing-library/react";
import App from "./App";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  Routes: () => <div data-component="Routes" />,
  useLocation: () => ({
    pathname: "",
  }),
}));

jest.mock("./components");

test("rendering required component", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
