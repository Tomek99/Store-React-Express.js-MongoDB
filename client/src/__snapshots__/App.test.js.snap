// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering required component 1`] = `
<div>
  <section
    class="columnWeb"
  >
    <div
      data-component="Routes"
    />
  </section>
</div>
`;
